.github/workflows/publish.yml

```diff
@@ -7,11 +7,50 @@ on:
 jobs:
   publish:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - run_command: "TARGET_OS=android TARGET_ARCH=arm64 make module"
+            canon_arch: "amd64"
+            platform: android/arm64
+          - run_command: "make module"
+            canon_arch: "arm64"
+            platform: linux/arm64
+          - run_command: "make module"
+            canon_arch: "amd64"
+            platform: linux/amd64
+
     steps:
     - uses: actions/checkout@v3
-    - uses: viamrobotics/build-action@v1
+
+    - name: Set up Go
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.21
+
+    - name: Install QEMU
+      run: |
+        sudo apt-get update
+        sudo apt install binfmt-support qemu-user-static
+
+    - name: Install canon
+      run: |
+        go install github.com/viamrobotics/canon@latest
+    
+    - name: Launch canon
+      run: |
+        canon -profile viam-rtsp-antique -arch ${{ matrix.canon_arch }}
+
+    - name: Build module
```

Comment:
I'm slightly suspicious that this might be running not in canon. Did you get around to verifying that all these executables run on their target archs?

Reply:
Good call! That would make sense - looking into it.

Reply:
Yep... arm64 upload is indeed x86_64 binary.

```
file /home/viam/.viam/packages/data/module/eacb3d64-4eab-43db-a9ef-e01e246f527a-viamrtsp-0_0_5-linux-arm64/bin/viamrtsp
/home/viam/.viam/packages/data/module/eacb3d64-4eab-43db-a9ef-e01e246f527a-viamrtsp-0_0_5-linux-arm64/bin/viamrtsp: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=30e4b5b9ecd4ba50ad2987e6429c0d85415eb144, for GNU/Linux 3.2.0, with debug_info, not stripped
```